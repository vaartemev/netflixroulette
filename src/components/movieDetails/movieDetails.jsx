import React from 'react';
import PropTypes from 'prop-types';
import { MovieTitle } from '../movieTitle/movieTitle';
import { Thumbnail } from '../thumbnail/thumbnail';

import './movieDetails.scss';

export const MovieDetails = ({
  movie: {
    poster_path,
    title,
    tagline,
    vote_average,
    overview,
    release_date,
    runtime
  }
}) => {
  return (
    <div className="about">
      <div className="about__picture">
        <Thumbnail src={poster_path} className="about__thumbnail" />
      </div>
      <div className="about__content">
        <MovieTitle className="about__title" title={title} />
        {vote_average !== 0 && (
          <div className="about__rating">{vote_average}</div>
        )}
        <h4 className="about__subtitle">{tagline}</h4>
        <div className="about__additional-info">
          <h5 className="release-date">
            {new Date(release_date).getFullYear()}
          </h5>
          {runtime && <h5 className="runtime">{runtime} min</h5>}
        </div>
        <p className="about__text">{overview}</p>
      </div>
    </div>
  );
};

MovieDetails.propTypes = {
  movie: PropTypes.shape({
    poster_path: PropTypes.string,
    title: PropTypes.string,
    vote_average: PropTypes.number,
    tagline: PropTypes.string,
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string
  })
};
