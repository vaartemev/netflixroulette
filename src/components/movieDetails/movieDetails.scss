.about {
	display: grid;
	grid-template-columns: 1fr 2fr;
	padding: 0 0 40px;
	margin-top: 20px;
	padding-right: 100px;
	color: #ffffff;
	position: relative;

	&__picture {
		width: 280px;
		height: 410px;
		overflow: hidden;
	}

	&__thumbnail {
		width: 100%;
	}

	&__title {
		color: #f55263;
		display: inline-block;
		font-size: 1.4em;
	}

	&__rating {
		display: inline-block;
		width: 40px;
		height: 40px;
		border: 1px solid #429442;
		color: #429442;
		margin-left: 10px;
		border-radius: 50%;
		line-height: 40px;
		text-align: center;
	}

	&__subtitle {
		font-weight: 300;
		color: #c7c7c7;
		margin: 10px auto;
	}

	&__additional {
		display: grid;
		grid-template-columns: 100px 100px;
		padding: 10px 0 0;
	}

	&__text {
		font-weight: 300;
		font-size: 16px;
		line-height: 24px;
		color: #fff;
	}
}


@media only screen and (max-width: 991px) {
	.about {
		padding: 0;
		margin-top: 40px;

		&__picture {
			padding-right: 20px;
			width: 240px;
		}
	}
}