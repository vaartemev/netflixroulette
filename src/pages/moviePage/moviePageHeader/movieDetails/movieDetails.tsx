/* eslint-disable camelcase */
import React from 'react';
import { MovieTitle, Thumbnail } from '../../../../components';

import './movieDetails.scss';

interface Types {
  poster_path: string;
  title: string;
  tagline: string;
  vote_average: number;
  overview: string;
  release_date: string;
  runtime: number;
}

interface MovieProps {
  movie: Types;
}

export const MovieDetails = ({
  movie: {
    poster_path,
    title,
    tagline,
    vote_average,
    overview,
    release_date,
    runtime,
  },
}: MovieProps) => {
  return (
    <div className="about">
      <div className="about__picture">
        <Thumbnail src={poster_path} className="about__thumbnail" />
      </div>
      <div className="about__content">
        <MovieTitle className="about__title" title={title} />
        {vote_average !== 0 && (
          <div className="about__rating">{vote_average}</div>
        )}
        <h4 className="about__subtitle">{tagline}</h4>
        <div className="about__additional-info">
          <h5 className="release-date">
            {new Date(release_date).getFullYear()}
          </h5>
          {runtime && <h5 className="runtime">{runtime} min</h5>}
        </div>
        <p className="about__text">{overview}</p>
      </div>
    </div>
  );
};
