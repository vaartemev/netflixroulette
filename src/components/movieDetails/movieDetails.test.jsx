import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MovieDetails } from './movieDetails';

describe('Movie Details', () => {
  const props = {
    movie: {
      poster_path: 'string1',
      title: 'string2',
      tagline: 'string3',
      vote_average: 7.4,
      overview: 'string5',
      release_date: '2018-03-30w',
      runtime: 2,
    },
  };

  const output = shallow(<MovieDetails {...props} />);

  it('renders correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
