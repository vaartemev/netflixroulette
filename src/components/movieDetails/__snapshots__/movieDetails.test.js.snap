// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details renders correctly 1`] = `
<div
  className="about"
>
  <div
    className="about__picture"
  >
    <Thumbnail
      className="about__thumbnail"
      src="string1"
    />
  </div>
  <div
    className="about__content"
  >
    <MovieTitle
      className="about__title"
      title="string2"
    />
    <div
      className="about__rating"
    >
      7.4
    </div>
    <h4
      className="about__subtitle"
    >
      string3
    </h4>
    <div
      className="about__additional-info"
    >
      <h5
        className="release-date"
      >
        NaN
      </h5>
      <h5
        className="runtime"
      >
        2
         min
      </h5>
    </div>
    <p
      className="about__text"
    >
      string5
    </p>
  </div>
</div>
`;
